newsView.js

```diff
@@ -0,0 +1,53 @@
+class NewsView {
+  constructor(newsModel, newsApi) {
+    this.newsModel = newsModel;
+    this.mainContainerEl = document.querySelector('#main-container');
+    this.buttonEl = document.querySelector('#search-button');
+    this.inputEl = document.querySelector('#search-input');
+    this.inputEl.value = '';
+
+    this.buttonEl.addEventListener('click', () => {
+      const searchTerm = document.querySelector('#search-input').value;
+
+      newsApi.getNews(searchTerm, (headlines) => {
+        newsModel.reset();
+        newsModel.addInfo(headlines);
+        this.displayNews();
+      });
+    });
+  }
+
+  displayNews() {
+    const headlines = this.newsModel.getHeadlines();
+    const links = this.newsModel.getLinks();
+    const images = this.newsModel.getImages();
+    const articlesArray = [];
+
+    headlines.forEach(headline => {
+      const newsEl = document.createElement('div');
```

Comment:
Maybe most of this could be wrapped in a private method called `createNewsElement`, so it would simplify the body of the loop — note that having all the elements of a single "news" centralised into a single object (as explained in my comment above) would also simplify the code:
```js
headlines.forEach(article => {
  const newsEl = createNewsElement(article);

  articlesArray.push(newsEl);
});
```

Reply:
Then you could refactor this private method further by adding more methods to construct each element, e.g `createArticleImgElement(article)`, `createArticleLinkElement(article)`, etc